build-system/tasks/unit.js

```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'using strict';
+
+const argv = require('minimist')(process.argv.slice(2));
+const babelify = require('babelify');
+const defaultConfig = require('./karma.conf');
+const log = require('fancy-log');
+const testConfig = require('../config');
+const {
+  createKarmaServer,
+  getAdTypes,
+  getBrowserConfig,
+  getUnitTestsToRun,
+  maybePrintArgvMessages,
+  maybeSetCoverageConfig,
+  refreshKarmaWdCache,
+  runTestInBatches,
+  startTestServer,
+} = require('./runtime-test/helpers');
+const {clean} = require('./clean');
+const {createCtrlcHandler, exitCtrlcHandler} = require('../ctrlcHandler');
+const {css} = require('./css');
+const {dist} = require('./dist');
+const {green, yellow, cyan, red} = require('ansi-colors');
+const {isTravisBuild} = require('../travis');
+
+function shouldNotRun() {
+  if (argv.local_changes && !getUnitTestsToRun(testConfig.unitTestPaths)) {
+    return true;
+  }
+
+  if (argv.saucelabs_lite) {
+    if (!process.env.SAUCE_USERNAME) {
+      throw new Error('Missing SAUCE_USERNAME Env variable');
+    }
+    if (!process.env.SAUCE_ACCESS_KEY) {
+      throw new Error('Missing SAUCE_ACCESS_KEY Env variable');
+    }
+  }
+
+  return false;
+}
+
+async function maybeBuild() {
+  if (argv.nobuild) {
+    return;
+  }
+
+  if (argv.local_changes) {
+    return css();
+  }
+
+  await clean();
+  await dist();
+}
+
+function setConfigOverrides(config) {
+  config.singleRun = !argv.watch && !argv.w;
+  config.client.mocha.grep = !!argv.grep;
+  config.client.verboseLogging = !!argv.verbose || !!argv.v;
+  config.client.captureConsole =
+    !!argv.verbose || !!argv.v || !!argv.local_changes;
+
+  config.browserify.configure = function(bundle) {
+    bundle.on('prebundle', function() {
+      log(green('Transforming tests with'), cyan('browserify') + green('...'));
+    });
+    bundle.on('transform', function(tr) {
+      if (tr instanceof babelify) {
+        tr.once('babelify', function() {
+          process.stdout.write('.');
+        });
+      }
+    });
+  };
+
+  // config.client is available in test browser via window.parent.karma.config
+  config.client.amp = {
+    useCompiledJs: false, // never compiled for unit tests
+    saucelabs: !!argv.saucelabs_lite,
```

Comment:
Deliberately leaving off `argv.saucelabs`?

Reply:
Yep because that's only used for integration tests